src/lib/transition_frontier/diff.ml

```diff
@@ -395,7 +402,7 @@ end) :
           else
             Some
               { user_commands= Breadcrumb.user_commands new_root
-              ; root_length= Some (1 + old_root_length) }
+              ; root_length= Some blockchain_length }
```

Comment:
I don't think this is correct. The root_length has nothing to do with the blockchain length, and is actually just a relative number to the root length. The semantics are meant to be: when you start a transition frontier, always initialize the root's "root_length" to 0 (or 1, I forget, doesn't matter though). Every new node you add, take it's parents "root_length" and add 1 to it. The property here is that you can take any block in the transition frontier and subtract it's root length from the root length of the root block and you will know the distance. Now, that said, this is basically obsolete now -- at one point it was required, but that was over a year ago when the interface to the block was different. We should just remove this field all together and always subtract the blockchain lengths, since the operation is will produce equivalent results. But, with the way the code is right now, I don't know if this code is fully safe since I'm not certain that the rest of the code expects the root length to operate this way.

Reply:
Nevermind, this is different from the Transition_frontier.Node.length and does not have the same dependencies in the code as that length does. My bad.